chat/models.py

```diff
@@ -0,0 +1,53 @@
+from django.conf import settings
+from django.db import models
+from django.dispatch import receiver
+
+
+def _get_service_choices():
+    return [(p, p) for p in settings.CHAT_SERVICES.keys()]
+
+
+class ChatRoom(models.Model):
+    service = models.CharField(max_length=32, choices=_get_service_choices())
+    name = models.CharField(max_length=255)
+
+    text_channel_id = models.CharField(max_length=255, null=True, blank=True)
+    voice_channel_id = models.CharField(max_length=255, null=True, blank=True)
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                name="unique_room_name_per_service", fields=["service", "name"]
+            ),
+        ]
+
+    def __str__(self):
+        return self.name
+
+    def get_service(self):
+        return settings.CHAT_SERVICES[self.service].get_instance()
+
+    def create_channels(self):
+        service = self.get_service()
+        self.text_channel_id = service.create_text_channel(self.name)
+        self.voice_channel_id = service.create_voice_channel(self.name)
+        self.save(update_fields=["text_channel_id", "voice_channel_id"])
+
+    def delete_channels(self):
```

Comment:
I'm assuming that other views / models should only be accessing ChatRoom and not ChatService, right? If so, we should add some comments specifying this.

If so, I think we may need to expose more here (deleting voice channel ONLY, e.g.) but I can do this later when I implement that feature.

Reply:
Sounds good. I'll update comments here and in the ChatService class to clarify that Django models/views should only use ChatRoom methods.